### DISCUSSION TITLE
Minio on xfs disk error: Metadata CRC error detected

### DISCUSSION THREAD

**Discussion by kietheros** [minio/minio#18953] [Q&A] [answered]
I have many individual servers, with each server hosting a single-node Minio installation on an HDD, **format xfs, noatime**. After a few months of operation, I encountered disk errors on three out of ten servers. The specific error messages were as follows:
```
[48544.666273] XFS: Metadata CRC error detected at xfs_agi_read_verify+0xce/0xf0 [xfs], xfs_agi block 0x10
[48544.666403] XFS: Unmount and run xfs_repair
```
Has anyone else experienced these errors?
Hard drives can experience issues over prolonged use, but 3 servers have same this xfs error is a bit abnormal. Does anyone think using ext4 might be better than xfs in terms of metadata corruption issues

**Comment by ravindk89** [comment]
> 3 servers have same this xfs error is a bit abnormal.

If the drives were from the same batch, it's not that abnormal. We've seen it before, especially with software like MinIO that can make maximum usage of a given drive. Maybe if you share a bit about the type of drive - manufacturer, product SKU, etc. it would be helpful.

> Does anyone think using ext4 might be better than xfs in terms of metadata corruption issues

Do not use EXT4. We have numerous issues with users running into inode exhaustion or similar problems on EXT4. Stick to XFS.

**Reply by kietheros** [reply]
I use many types of disks.
Info of failed disks:
```
Model Family:     Seagate IronWolf Pro
Device Model:     ST16000NE000-2RW103
Serial Number:    ZL2HPSFK

Model Family:     Seagate IronWolf Pro
Device Model:     ST16000NE000-2RW103
Serial Number:    ZL2HPQVC

Device Model:     TRM S100 512GB 
Serial Number:    UCFW16778F
Firmware Version: U0430A0
```

**Reply by kietheros** [reply]
Each Minio storage holds 30M files, each with a size of 50KB. The data is read at a frequency of around 30,000 requests per day, which is relatively low. Given that persistent data assurance isn't crucial, I am planning to tune Minio as follows to mitigate excessive disk activity from the minio scanner:

- Set the scanner speed to the slowest setting
- Disable expiration rules: I put files by date and use a `mc rm date/` for file deletion.
What are your thoughts on this approach?"

**Reply by harshavardhana** [reply, marked as answer]
We do not do any architectural debugging as community support; if you want a more white glove service, you should sign up for a subscription with us.  We can guide you on how to build a successful deployment for your workload.

**Reply by kietheros** [reply]
Thank you for your feedback. We are also facing various challenges when self-hosting object storage, and we will consider the minio standard license and s3.

**Reply by klauspost** [reply]
In overall terms `XFS: Metadata CRC error detected at xfs_agi_read_verify+0xce/0xf0 [xfs], xfs_agi block 0x10` happens completely outside MinIO - so you have an OS/Filesystem/Hardware problem.

**Reply by kietheros** [reply]
@klauspost I undertand that. I'm trying to find a workaround by setting minio scanner speed to slowest, delay 1000x, cycle=1000s. But it seems to still be running quickly.
